src/inc-lib-init.php

```diff
@@ -567,20 +567,114 @@ function lovd_getVariantInfo ($sVariant, $sTranscriptID = '')
 
             // (int) to get rid of the '+' if it's there.
             $aResponse['position_start_intron'] = (int) $sStartPositionIntron;
-            $aResponse['position_end_intron'] = (int) ($sEndPositionIntron? $sEndPositionIntron : $sStartPositionIntron);
+            $aResponse['position_end_intron'] = (int) ($sEndPosition? $sEndPositionIntron : $sStartPositionIntron);
         }
 
-        // Variant type.
-        if (preg_match('/^[ACGT]>[ACGT]$/', $sVariant)) {
-            $aResponse['type'] = 'subst';
+
+
+    // If that didn't work, try matching variants with uncertain positions.
+    // We're not super picky, and don't check the end of the variant.
+    } elseif (preg_match('/^([cgmn])\.\((\d+|\?)([-+](?:\d+|\?))?_(\d+|\?)([-+](?:\d+|\?))?\)_\((\d+|\?)([-+](?:\d+|\?))?_(\d+|\?)([-+](?:\d+|\?))?\)(d(el|up)|(inv|ins))/', $sVariant, $aRegs)) {
+        //                   1 = Prefix; indicates what kind of positions we can expect, and what we'll output.
+        //                               2 = Earliest start position, might be a question mark.
+        //                                       3 = Earlier start position intronic offset, if available.
+        //                                                         4 = Latest start position, might be a question mark.
+        //                                                                 5 = Latest start position intronic offset, if available.
+        //                                                                                       6 = Earliest end position, might be a question mark.
+        //                                                                                               7 = Earliest end position intronic offset, if available.
+        //                                                                                                                  8 = Latest end position, might be a question mark.
+        //                                                                                                                          9 = Latest end position intronic offset, if available.
+        //                                                                                                                                              10 = The variant, which we'll use to determine the type.
+
+        list(, $sPrefix, $sStartPositionEarly, $sStartPositionEarlyIntron, $sStartPositionLate, $sStartPositionLateIntron, $sEndPositionEarly, $sEndPositionEarlyIntron, $sEndPositionLate, $sEndPositionLateIntron, $sVariant) = $aRegs;
+
+        // Store positions.
+        // If each position (start, end) has two numeric positions, we choose the
+        //  middle one (latest start, earliest end). Otherwise, we pick the numeric one.
+        // We do require at least one numeric start position and one numeric end position.
```

Comment:
Johan has requested to take the average of the two numeric positions, and not the middle ones.